src/uu/ls/src/ls.rs

```diff
@@ -1224,12 +1259,53 @@ impl PathData {
             None => OnceCell::new(),
         };
 
+        let substitute_string = "?".to_string();
+        let security_context = if config.context {
+            if config.selinux_supported {
+                #[cfg(feature = "selinux")]
+                {
+                    match selinux::SecurityContext::of_path(&p_buf, must_dereference, false) {
```

Comment:
what do you think about moving this block into a new function to simplify a bit the code?

Reply:
Sounds good. Done.